Update to README.md to be more explanative (#635)
absa-subatomic/quantum-mechanic: README.md

An Atomist [automation client](https://github.com/atomist/automation-client-ts)
with command and event handlers for integration between various external infrastructure components.

Quantum Mechanic extends on the Atomist Automation Client for Slack. It contains command and event handlers for 

> OpenShift and leverages of the Gluon project for auditing and context awareness.

Reviewer: ```suggestion
OpenShift and leverages off the Gluon project for auditing and context awareness.
```
I think that is correct English?

Author: apparently off is the opposite of on, or can be used to indicate away from a place. i.e the dog rand off, your are way off etc.

> OpenShift and leverages of the Gluon project for auditing, context awareness as well as queing.

Reviewer: We don't really do any queuing through Gluon.

> Therefore, follow read the instructions for [Running the Automation Client](https://github.com/atomist/automation-client-ts#running-the-automation-client)

Author: changed to follow

from the [`automation-client-ts`](https://github.com/atomist/automation-client-ts) GitHub repository to gain a better
understanding of the library.

## Development setup

### 1. Install Node and NPM 

Download and install node - v8.12.0 and npm - v6.4.1 or higher from the [NodeJS](https://nodejs.org/en/) site

### 2. Clone the repo

Use the following command in a terminal to clone the repository to the desired local directory.

`git clone https://github.com/absa-subatomic/quantum-mechanic.git`

### 3. Install the Node dependencies

Navigate to the folder you have cloned and use the following node package manager command to install dependencies

`npm install`

### 4. Local configuration

> The configuration is drawn from the `atomist.config.ts` file by default, however provision for local development has 

Reviewer: ```suggestion
The Atomist Automation client configuration is drawn from the `atomist.config.ts` file, however the Quantum Mechanic specific configuration can be added to a `local.json` file. 
```

> been made with the use of a local.json.

Reviewer: ```suggestion
```

> You will need to create a new `config/local.json` file. This file can provide local configuration values over and above

Author: as I understand it, the atomist.config.ts is the default. If a local.json exist it overrides. So it can be added? 

I think in production we dont use the local.json. Hence why I said it this way... Needs improvement?

> You will need to create a new `config/local.json` file. This file can provide local configuration values over and above

Reviewer: We do use the `local.json`(does not need to be called this). The `atomist.config.ts` and the `local.json` config files house separate configuration information. `atomist.config.ts` is literally Atomist stuff. It is what Atomist needs for the Automation client to actually run. There are some fields in this that are pulled from our internal config. `local.json` houses the Quantum Mechanic/Subatomic configuration. This is our config - i.e. the stuff that we use and reference in our code. It gets loaded into this class here: https://github.com/absa-subatomic/quantum-mechanic/blob/02332c589438f9f0b9d4c2d49f1b7dbdb6ea43e3/src/config/QMConfig.ts#L13. You can see how it is used for example here: https://github.com/absa-subatomic/quantum-mechanic/blob/02332c589438f9f0b9d4c2d49f1b7dbdb6ea43e3/src/gluon/services/gluon/MemberService.ts#L23

> You will need to create a new `config/local.json` file. This file can provide local configuration values over and above

Author: Thanks for clearing that up.

```console
$ npm run test
```

## Contributing
Please read [CONTRIBUTING.md](https://gist.github.com/PurpleBooth/b24679402957c63ec426) for details on our code of conduct, and the process for submitting pull requests to us.

## Versioning
No versioning is in place at the time of creating this documentation

## Authors
A bunch of dudes, at a time, in a place. Sub-Atomic team

## License
This project is licensed under the Apache License v2.0 - see the LICENSE file for details

## Acknowledgements

> Hat tips to anyone inspiration

Author: Leave the hook or keep it clean, agreed, I can comment it out as well
